core/system/src/mpu/vmpu_freescale_k64_mem.c

```diff
@@ -33,19 +57,61 @@ typedef struct
     uint32_t count;
 } TBoxACL;
 
-uint32_t g_mem_acl_count, g_mem_acl_user;
+uint32_t g_mem_acl_count, g_mpu_slot;
 static TMemACL g_mem_acl[UVISOR_MAX_ACLS];
 static TBoxACL g_mem_box[UVISOR_MAX_BOXES];
 
+int vmpu_mem_push_page_acl(uint32_t start_addr, uint32_t end_addr)
+{
+    TMemACL rgd;
+    volatile uint32_t* word;
+
+    /* Check that start and end address are aligned to 32-byte. */
+    if (start_addr & 0x1F || end_addr & 0x1F) {
+        return -1;
+    }
+
+    rgd.word[0] = start_addr;
+    rgd.word[1] = end_addr;
+    /* Permissions are fixed to only allow unprivileged write/read/execute. Duh. */
+    rgd.word[2] = 0x1E;
+
+    /* Insert into MPU regions in RR fashion. */
+    word = MPU->WORD[g_mpu_slot];
```

Comment:
Freescale does _not_ provide a [proper struct for this access](https://github.com/ARMmbed/uvisor/blob/master/core/cmsis/inc/mpu_kinetis.h#L1124). I will add a type for this.